modules/sfp_badpackets.py

```diff
@@ -0,0 +1,242 @@
+            self.errorState = True
+            return None
+
+        # Don't look up stuff twice
+        if eventData in self.results:
+            self.sf.debug("Skipping " + eventData + " as already mapped.")
+            return None
+        else:
+            self.results[eventData] = True
+
+        if eventName == 'NETBLOCK_OWNER':
+            if not self.opts['netblocklookup']:
+                return None
+            else:
+                if IPNetwork(eventData).prefixlen < self.opts['maxnetblock']:
+                    self.sf.debug("Network size bigger than permitted: " +
+                                str(IPNetwork(eventData).prefixlen) + " > " +
+                                str(self.opts['maxnetblock']))
+                    return None
+        
+        if eventName == 'NETBLOCK_MEMBER':
+            if not self.opts['subnetlookup']:
+                return None
+            else:
+                if IPNetwork(eventData).prefixlen < self.opts['maxsubnet']:
+                    self.sf.debug("Network size bigger than permitted: " +
+                                str(IPNetwork(eventData).prefixlen) + " > " +
+                                str(self.opts['maxsubnet']))
+                    return None
+                    
+        qrylist = list()
+        if eventName.startswith("NETBLOCK_"):
+            for ipaddr in IPNetwork(eventData):
+                qrylist.append(str(ipaddr))
+                self.results[str(ipaddr)] = True
+        else:
+            # If user has enabled affiliate checking
+            if eventName == "AFFILIATE_IPADDR" and not self.opts['checkaffiliates']:
+                return None
+            qrylist.append(eventData)
+        
+        for addr in qrylist:
+
+            nextPageHasData = True    
+            if self.checkForStop():
+                return None
+            
+            currentOffset = 0
+            while nextPageHasData:
+                data = self.queryIPAddress(addr, currentOffset)
+
+                if data is None:
+                    nextPageHasData = False
+                    break
+
+                count = data.get('count')
+                if count is None or int(count) == 0:
+                    nextPageHasData = False
+                    break
+
+                # Data is reported about the IP Address
+                if eventName.startswith("NETBLOCK_"):
+                    ipEvt = SpiderFootEvent("IP_ADDRESS", addr, self.__name__, event)
+                    self.notifyListeners(ipEvt)
+
+                results = data.get('results')
+                if results is None:
+                    nextPageHasData = False
+                    break
+
+                if results:
+                    evt = SpiderFootEvent("RAW_RIR_DATA", str(results), self.__name__, event)
+                    self.notifyListeners(evt)
+                    
+                    for result in results:
+                        maliciousIP = result.get('source_ip_address')
```

Comment:
Any reason why this would be different from the IP address queried? We obviously don't want to land in a situation where the `IP_ADDRESS` event IP is different from the one stated in the `MALICIOUS_IPADDR` event.